app/src/main/java/org/kiwix/kiwixmobile/language/LanguageScreen.kt

```diff
@@ -0,0 +1,214 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import org.kiwix.kiwixmobile.core.R
+import org.kiwix.kiwixmobile.language.composables.LanguageListItem.HeaderItem
+import org.kiwix.kiwixmobile.language.composables.LanguageListItem.LanguageItem
+import org.kiwix.kiwixmobile.language.composables.LanguageItemRow
+import org.kiwix.kiwixmobile.language.composables.TopBar
+import org.kiwix.kiwixmobile.language.viewmodel.Action
+import org.kiwix.kiwixmobile.language.viewmodel.Action.Select
+import org.kiwix.kiwixmobile.language.viewmodel.LanguageViewModel
+import org.kiwix.kiwixmobile.language.viewmodel.State
+import org.kiwix.kiwixmobile.language.viewmodel.State.Content
+import org.kiwix.kiwixmobile.ui.theme.KiwixTheme
+
+@Suppress("all")
+@SuppressLint("UnusedMaterial3ScaffoldPaddingParameter")
+@Composable
+fun LanguageScreen(
+  languageViewModel: LanguageViewModel,
+  onNavigationClick: () -> Unit
+) {
+  val state by languageViewModel.state.observeAsState(State.Loading)
+  val listState: LazyListState = rememberLazyListState()
+  val context = LocalContext.current
+  val coroutineScope: CoroutineScope = rememberCoroutineScope()
+  var searchText by remember { mutableStateOf("") }
+  var isSearchActive by remember { mutableStateOf(false) }
+  var updateListState by remember { mutableStateOf(false) }
+
+  KiwixTheme {
+    Scaffold(
+      topBar = {
+        TopBar(
+          title = stringResource(id = R.string.select_languages),
+          searchText = searchText,
+          isSearchActive = isSearchActive,
+          onNavigationClick = onNavigationClick,
+          onSearchClick = { isSearchActive = true },
+          onBackClick = {
+            if (isSearchActive) {
+              isSearchActive = false
+              searchText = ""
+              languageViewModel.actions.offer(Action.Filter(searchText))
+            }
+          },
+          onSearchTextSubmit = {
+            languageViewModel.actions.offer(Action.SaveAll)
+            updateListState = true
+          },
+          onSearchTextClear = {
+            searchText = ""
+            languageViewModel.actions.offer(Action.Filter(searchText))
+          },
+          onSearchTextChanged = {
+            searchText = it
+            languageViewModel.actions.offer(Action.Filter(it))
+          }
+        )
+      }
+    ) {
+      Column(modifier = Modifier.fillMaxSize()) {
+        // spacer to account for top app bar
+        Spacer(modifier = Modifier.height(56.dp))
+        when (state) {
+          State.Loading -> {
+            LoadingIndicator()
+          }
+
+          is Content -> {
+            LazyColumn(
```

Comment:
We should split this long method into a separate method(smaller and focused ones) for better reliability, and maintainability. It is a little bit hard to understand/maintain what is going on in this method.

E.g. for header text, there should be a separate composable function named: `HeaderText`, and the styling for HeaderText would be in that function.